src/core/public/analytics/track_clicks.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { fromEvent } from 'rxjs';
+import type { AnalyticsClient } from '@kbn/analytics-client';
+
+/** HTML attributes that should be skipped from reporting because they might contain user data */
+const POTENTIAL_PII_HTML_ATTRIBUTES = ['value'];
+
+/**
+ * Registers the event type "click" in the analytics client.
+ * Then it listens to all the "click" events in the UI and reports them with the `target` property being a
+ * full list of the element's and its parents' attributes. This allows
+ * @param analytics
+ */
+export function trackClicks(analytics: AnalyticsClient, isDevMode: boolean) {
+  analytics.registerEventType<{ target: string[] }>({
+    eventType: 'click',
+    schema: {
```

Comment:
I thought FullStory already had its own internal collection of click events? If that's the case, maybe we should limit the event to the teleV3 shipper?

Reply:
https://github.com/elastic/kibana/issues/131146 already limits the events that are sent to FullStory.